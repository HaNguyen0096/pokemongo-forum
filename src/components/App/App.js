import React, { Component } from 'react'
import { Route, Switch } from 'react-router-dom'
import Header from '../Header/Header'
import MainPage from '../../routes/MainPage/MainPage'
import TopicPage from '../../routes/TopicPage/TopicPage'
import ThreadPage from '../../routes/ThreadPage/ThreadPage'
import LoginPage from '../../routes/LoginPage/LoginPage'
import RegistrationPage from '../../routes/RegistrationPage/RegistrationPage'
import NotFoundPage from '../../routes/NotFoundPage/NotFoundPage'
//import ThreadContext from '../../contexts/ThreadContext'
//import config from '../../config'

//import Footer from '../Footer/Footer';
import './App.css'


class App extends Component {

  state = { 
    hasError: false 
  }

  static getDerivedStateFromError(error) {
    console.error(error)
    return { hasError: true }
  }

  render() {
    return(
      < >
        <div className='App'>
        <header className='App__header'>
          <Header />
        </header>
        <main className='App__main'>
        {this.state.hasError && <p className='red'>There was an error! Oh no!</p>}
          <Switch>
            <Route 
              exact
              path={'/'}
              component={MainPage}
            />
            <Route
              exact
              path={'/login'}
              component={LoginPage}
            />
            <Route
              exact
              path={'/register'}
              component={RegistrationPage}
            />
            <Route
              exact
              path={'/topic/:topicId'}
              component={TopicPage}
            />
            <Route
              path={'/topic/:topicId/:threadId'}
              component={ThreadPage}
            />           
            <Route
              component={NotFoundPage}
            />
          </Switch>
        </main>        
      </div>
      </>     
    )
  }
}

export default App;
