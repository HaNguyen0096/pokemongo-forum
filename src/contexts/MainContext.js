import React, {Component} from 'react'

const MainContext = React.createContext({
  threadList: [],
  topicList: [],
  error: null,
  setError: () => {},
  clearError: () => {},
  setTopicList: () => {},
  //addThread: () => {},
  //deleteThread: () => {},
})

export default MainContext

export class MainProvider extends Component {

  state = {
    threadList : [],
    topicList : [],
    error: null,
  };

  setTopicList = topicList => {
    this.setState({topicList})
  }

  setThreadList = threadList => {
    console.log(threadList)
    this.setState({threadList})
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render(){
    const value = {
      threadList: this.state.threadList,
      topicList: this.state.topicList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setTopicList: this.setTopicList,
      setThreadList: this.setThreadList,
    }
    return (
      <MainContext.Provider value={value}>
        {this.props.children}
      </MainContext.Provider>
    )
  }
}