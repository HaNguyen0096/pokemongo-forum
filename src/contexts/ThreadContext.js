import React, {Component} from 'react'

const ThreadContext = React.createContext({
  thread: null,
  comments: [],
  error: null,
  setError: () => {},
  clearError: () => {},
  setThread: () => {},
  clearThread: () => {},
  setComments: () => {},
  addComment: () => {},
  deleteComment: () => {},
  deleteThread: () => {},

})

export default ThreadContext

export class ThreadProvider extends Component {

  state = {
    thread: null,
    error: null,
  };

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  setThread = thread => {
    this.setState({ thread })
  }

  clearThread = () => {
    this.setThread(null)
    this.setComments([])
  }

  setComments = comments => {
    this.setState({ comments })
  }

  addComment = comment => {
    this.setComments([
      ...this.state.comments,
      comment
    ])
  }

  render(){
    const value = {
      thread: this.state.thread,
      comments: this.state.comments,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setThread: this.setThread,
      setComments: this.setComments,
      clearThread: this.clearThread,
      addComment: this.addComment,
      deleteComment: this.deleteComment,
      deleteThread: this.deleteThread,
    }
    return (
      <ThreadContext.Provider value={value}>
        {this.props.children}
      </ThreadContext.Provider>
    )
  }
}