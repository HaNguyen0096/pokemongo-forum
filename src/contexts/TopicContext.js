import React, {Component} from 'react'

const TopicContext = React.createContext({
  topic: null,
  threads: [],
  comments: [],
  error: null,
  setError: () => {},
  clearError: () => {},
  setTopic: () => {},
  clearTopic: () => {},
  setThreads: () => {},
  addThread: () => {},
  setComments: () => {},
})

export default TopicContext

export class TopicProvider extends Component {

  state = {
    topic: null,
    error: null,
  };

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  setTopic = topic => {
    this.setState({ topic })
  }

  setThreads = threads => {
    this.setState({ threads })
  }

  setComments = comments => {
    this.setState({comments})
  }

  clearTopic = () => {
    this.setTopic(null)
    this.setThreads([])
  }

  addThread = thread => {
    this.setThreads([
      ...this.state.threads,
      thread
    ])
  }

  render(){
    const value = {
      topic: this.state.topic,
      threads: this.state.threads,
      comments: this.state.comments,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setTopic: this.setTopic,
      setThreads: this.setThreads,
      clearTopic: this.clearTopic,
      addThread: this.addThread,
      setComments: this.setComments,
    }
    return (
      <TopicContext.Provider value={value}>
        {this.props.children}
      </TopicContext.Provider>
    )
  }
}