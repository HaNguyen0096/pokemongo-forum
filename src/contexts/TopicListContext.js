import React, { Component } from 'react'

const TopicListContext = React.createContext({
  topicList: [],
  threadList: [],
  error: null,
  setError: () => {},
  clearError: () => {},
  setTopicList: () => {},
  setThreadList: () => {},
})

export default TopicListContext

export class TopicListProvider extends Component {
  state = {
    topicList: [],
    threadList: [],
    error: null,
  };

  setTopicList = topicList => {
    this.setState({topicList})
  }

  setThreadList = threadList => {
    this.setState({threadList})
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      topicList: this.state.topicList,
      threadList: this.state.threadList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setTopicList: this.setTopicList,
      setThreadList: this.setThreadList,
    }
    return (
      <TopicListContext.Provider value={value}>
        {this.props.children}
      </TopicListContext.Provider>
    )
  }
}
