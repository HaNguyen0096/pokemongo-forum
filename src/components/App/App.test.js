import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import { BrowserRouter } from 'react-router-dom'
import { TopicProvider } from '../../contexts/TopicContext'
import { TopicListProvider } from '../../contexts/TopicListContext'
import { ThreadProvider } from '../../contexts/ThreadContext'

it('renders App without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter>
    <TopicListProvider>
      <TopicProvider>
        <ThreadProvider>
          <App />
        </ThreadProvider>
      </TopicProvider>
    </TopicListProvider>
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

