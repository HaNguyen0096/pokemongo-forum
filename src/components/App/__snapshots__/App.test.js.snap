// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App by default 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <TopicListProvider>
    <TopicProvider>
      <ThreadProvider>
        <App />
      </ThreadProvider>
    </TopicProvider>
  </TopicListProvider>
</Router>
`;

exports[`App component renders the UI for App as expected 1`] = `
<div
  className="App"
>
  <header
    className="App__header"
  >
    <div
      className="header"
    >
      <div
        className="headerTop"
      >
        <img
          alt="Pokemon GO forum"
          className="banner"
          src="go-banner.png"
        />
        <div
          className="loginNav"
        >
          <div
            className="title"
          >
            Pokemon GO Forum
          </div>
        </div>
      </div>
      <nav
        className="navBar"
      />
    </div>
  </header>
  <main
    className="App__main"
  >
    <div
      className="MainPage"
    >
      <section
        className="Section"
      >
        <div
          className="Welcome_Message"
        >
          <p>
            Welcome to the forum, feel free to ask any question!
          </p>
        </div>
      </section>
    </div>
  </main>
</div>
`;

exports[`TopicList component renders CommentList by default 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <TopicListProvider>
    <TopicProvider>
      <ThreadProvider>
        <App />
      </ThreadProvider>
    </TopicProvider>
  </TopicListProvider>
</Router>
`;

exports[`TopicList component renders the UI for CommentList as expected 1`] = `
<div
  className="App"
>
  <header
    className="App__header"
  >
    <div
      className="header"
    >
      <div
        className="headerTop"
      >
        <img
          alt="Pokemon GO forum"
          className="banner"
          src="go-banner.png"
        />
        <div
          className="loginNav"
        >
          <div
            className="title"
          >
            Pokemon GO Forum
          </div>
        </div>
      </div>
      <nav
        className="navBar"
      />
    </div>
  </header>
  <main
    className="App__main"
  >
    <div
      className="MainPage"
    >
      <section
        className="Section"
      >
        <div
          className="Welcome_Message"
        >
          <p>
            Welcome to the forum, feel free to ask any question!
          </p>
        </div>
      </section>
    </div>
  </main>
</div>
`;
